Cabal-syntax/src/Distribution/Types/UnqualComponentName.hs

```diff
@@ -91,3 +91,21 @@ packageNameToUnqualComponentName = UnqualComponentName . unPackageNameST
 -- @since 2.0.0.2
 unqualComponentNameToPackageName :: UnqualComponentName -> PackageName
 unqualComponentNameToPackageName = mkPackageNameST . unUnqualComponentNameST
+
+-- | Combine names in targets (partial function). Useful in 'Semigroup'
+-- and similar instances.
+combineName :: a -> a -> (a -> UnqualComponentName) -> String ->
```

Comment:
This should probably be called `combineNames`? I know it doesn't really combine different names, only one name, but this may be too clever for its own good.

Also, I wish Haskell ecosystem had a guideline for naming partial functions (like a trailing `!` or `?` but alas these wouldn't lex).

Reply:
`combineNameUnsafe`? But then it would imply there is a `combineNameSafe` version…

Reply:
Dunno. Since, there's no precedent (I'm aware of) and nothing pops up immediately, I would just do `combineNames`, perhaps.

A crazy idea: `combineNamesAffine`: it's 0 names (both empty), 1 name, or error...

It's just the singular case in the name that threw my off. Anything more than that I leave to you!